src/db.py

```diff
@@ -0,0 +1,93 @@
+"""
+Supabase Database Interaction Script
+
+This script demonstrates how to interact with a Supabase database using Python.
+It utilises the `dotenv` library to load environment variables
+from a `.env` file and the `supabase-py` library to perform database operations.
+The script includes functions to write data to and read data from a Supabase
+table.
+
+Environment Variables:
+- SUPABASE_URL: The URL of the Supabase instance.
+- SUPABASE_KEY: The API key for accessing the Supabase instance.
+
+"""
+
+import os
+import platform
+
+from dotenv import load_dotenv
+from supabase import create_client, Client
+
+# Load environment variables from .env file
+load_dotenv()
+
+# Project Settings -> API
+SUPABASE_URL = os.getenv("SUPABASE_URL")
+
+# Project Settings -> API -> Project API keys
+SUPABASE_KEY = os.getenv("SUPABASE_KEY")
+
+# Initialising the Supabase client
+supabase: Client = create_client(SUPABASE_URL, SUPABASE_KEY)
+
+
+def write_to_db(
+    hosting_provider, repo_name, file_path, guessed_language, hostname=None
+):
+    """
+    Writes a new record to the 'guessed_languages' table in the Supabase
+    database.
+    Parameters:
+        hosting_provider (str): The hosting provider of the repository.
+        repo_name (str): The name of the repository.
+        file_path (str): The file path in the repository.
+        guessed_language (str): The language guessed for the file.
+        hostname (str, optional): The hostname of the machine. Defaults to the
+        current platform's node name.
+    """
+    if hostname is None:
+        hostname = platform.node()
+    try:
+        # Define data to be inserted
+        data = {
+            "hosting_provider": hosting_provider,
+            "repo_name": repo_name,
+            "file_path": file_path,
+            "guessed_language": guessed_language,
+            "hostname": hostname,
+        }
+        supabase.table("guessed_languages").insert(data).execute()
+
+        print("Data inserted successfully")
+
+    except Exception as error:
+        print(f"Error: {error}")
+
+
+def read_from_db():
+    """
+    Reads and returns distinct file paths from the 'guessed_languages' table in
+    the Supabase database.
+
+    Returns:
+        set: A set of distinct file paths.
+    """
+    try:
+        # Query the distinct file_path values from the guessed_languages table
+        response = supabase.table("guessed_languages").select("file_path").execute()
+
+        # Extract the file_path values and return them as a set
+        file_paths = set(record["file_path"] for record in response.data)
+
+        return file_paths
+
+    except Exception as error:
```

Comment:
- Improved `read_from_db` function to handle empty database and enhance error logging

It would be a way of signalling that no data could be retrieved.  I made sure that the logging captures the details of the exception for debugging purposes. Please advise if you would like me to remove this line.

Reply:
Thanks for your feedback.

I removed it. Returning an empty set in the case of an exception could indeed mask the fact that something went wrong . I have re-raised the exception after logging it, in the event of an actual error (e.g., connectivity issues, SQL syntax error, etc.), it will not silently fail.